src/Microsoft.Azure.WebJobs.Host/Executors/JobHostConfigurationExtensions.cs

```diff
@@ -161,12 +162,34 @@ public static ServiceProviderWrapper CreateStaticServices(this JobHostConfigurat
             INameResolver nameResolver = services.GetService<INameResolver>();
             IExtensionRegistry extensions = services.GetExtensions();
             IStorageAccountProvider storageAccountProvider = services.GetService<IStorageAccountProvider>();
+            ILoggerFactory loggerFactory = services.GetService<ILoggerFactory>();
+            IFunctionResultAggregatorFactory aggregatorFactory = services.GetService<IFunctionResultAggregatorFactory>();
+            IAsyncCollector<FunctionInstanceLogEntry> functionEventCollector = null;
+
+            // Create the aggregator if all the pieces are configured
+            IAsyncCollector<FunctionInstanceLogEntry> aggregator = null;
+            if (loggerFactory != null && aggregatorFactory != null && config.Aggregator.IsEnabled)
+            {
+                aggregator = aggregatorFactory.Create(config.Aggregator.BatchSize, config.Aggregator.FlushTimeout, loggerFactory);
+            }
 
             IQueueConfiguration queueConfiguration = services.GetService<IQueueConfiguration>();
             var blobsConfiguration = config.Blobs;
 
             TraceWriter trace = services.GetService<TraceWriter>();
-            IAsyncCollector<FunctionInstanceLogEntry> fastLogger = services.GetService<IAsyncCollector<FunctionInstanceLogEntry>>();                               
+            IAsyncCollector<FunctionInstanceLogEntry> registeredFunctionEventCollector = services.GetService<IAsyncCollector<FunctionInstanceLogEntry>>();
+
+            if (registeredFunctionEventCollector != null && aggregator != null)
+            {
+                // If there are both an aggregator and a registered FunctionEventCollector, wrap them in a composite
+                functionEventCollector = new CompositeFunctionEventCollector(new[] { registeredFunctionEventCollector, aggregator });
```

Comment:
>CompositeFunctionEventCollector [](start = 45, length = 31)

Can this be in Script (ie, up the stack)? IE, the SDK is a produce that just emits events to the registered IAsyncCollector<FunctionInstanceLogEntry>. 
Script is a consumer that pulls these events and does whatever it needs (pushes to fast logger; pushes to AppInsight; etc)